lib/semian/net_http.rb

```diff
@@ -67,9 +76,11 @@ def reset_exceptions
     Semian::NetHTTP.reset_exceptions
 
     def raw_semian_options
-      options = semian_configuration
-      options = options.dup unless options.nil?
-      options
+      check_semian_configuration_changed
+      @raw_semian_options ||= begin
+        @raw_semian_options = Semian::NetHTTP.retrieve_semian_configuration(address, port)
+        @raw_semian_options = @raw_semian_options.dup unless @raw_semian_options.nil?
+      end
```

Comment:
I don't understand why we can justify caching these values at all, since the options can change due to change to the function that could use time, random, .. or something else to spit out different values at different times:

```ruby
Semian::NetHTTP.semian_configuration = proc { |host, port|
  if Time.now.hour > 8 && Time.now.hour < 18 # during peak hours
    SEMIAN_PARAMETERS.merge(tickets: 10, name: "#{host}_#{port}")
  else # during the night
   SEMIAN_PARAMETERS.merge(tickets: 5, name: "#{host}_#{port}")
  end
}
```

Why do we need to cache it in the first place?